### PR TITLE
Challenge 3 - John Kim

### PR DESCRIPTION
**Pull request by abcdefghijohn**
x__x

### CODE REVIEW

**Comment by KathleenScriver** [comment]
Let's get into the habit of committing more frequently to document your workflow. After each test, you'll want to commit, then when you get that test passing you'll want to commit again.

File: lib/apartment.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Apartment
+
+  attr_reader :number, :monthly_rent, :bathrooms, :bedrooms
+
+  def initialize(apartment_info)
+    @number = apartment_info[:number]
+    @monthly_rent = apartment_info[:monthly_rent]
+    @bathrooms = apartment_info[:bathrooms]
+    @bedrooms = apartment_info[:bedrooms]
```
Comment (KathleenScriver) [inline lib/apartment.rb:9]: Great job working with a hash argument here.

Where is your attribute and method for #renter? 

You're missing the last two methods of iteration ONE.

File: test/apartment_renter_test.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/renter'
+require './lib/apartment'
+
+class ApartmentTest < Minitest::Test
+
+  def test_it_exists
+  renter1 = Renter.new("Jessie")
+  assert_instance_of Renter, renter1
```
Comment (KathleenScriver) [inline test/apartment_renter_test.rb:10]: Testing a Renter should be in the RenterTest, not the ApartmentTest. Each class should have it's own test file.

File: test/apartment_renter_test.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  def test_it_has_a_name
+    renter1 = Renter.new("Jessie")
+    assert_equal "Jessie", renter1.name
```
Comment (KathleenScriver) [inline test/apartment_renter_test.rb:15]: Again, this should be in a RenterTest file.

File: test/apartment_renter_test.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      bedrooms: 1})
+      assert_equal 1200, unit1.monthly_rent
+    end
+
+    def test_apartment_has_a_bathroom
+      unit1 = Apartment.new({number: "A1",
+      monthly_rent: 1200, bathrooms: 1,
+      bedrooms: 1})
+      assert_equal 1, unit1.bathrooms
+    end
```
Comment (KathleenScriver) [inline test/apartment_renter_test.rb:43]: Great, you're missing the test for the bedrooms attribute though.

File: test/building_test.rb [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      monthly_rent: 1200, bathrooms: 1,
+      bedrooms: 1})
+    unit2 =
+      Apartment.new({number: "B2",
+      monthly_rent: 999, bathrooms: 2,
+      bedrooms: 2})
+      building.add_unit(unit1)
+      building.add_unit(unit2)
+
+      assert_equal [unit1, unit2] , building.units
```
Comment (KathleenScriver) [inline test/building_test.rb:33]: Great job testing the initial state in the test above, and now this changed state of #units.

File: test/building_test.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    building = Building.new
+    unit1 =
+      Apartment.new({number: "A1",
+      monthly_rent: 1200, bathrooms: 1,
+      bedrooms: 1})
+    unit2 =
+      Apartment.new({number: "B2",
+      monthly_rent: 999, bathrooms: 2,
+      bedrooms: 2})
+    unit2 =
```
Comment (KathleenScriver) [inline test/building_test.rb:85]: Watch your variables here. You have three variables named `unit2` so each one will override the next.

File: test/building_test.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      Apartment.new({number: "D4",
+      monthly_rent: 1500, bathrooms: 2,
+      bedrooms: 3})
+      renter1 = Renter.new("Spencer")
+
+      building.add_unit(unit1)
+      building.add_unit(unit2)
+      building.add_unit(unit3)
```
Comment (KathleenScriver) [inline test/building_test.rb:97]: You have not defined a variable `unit3`

File: test/building_test.rb [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+      assert_equal [], building.rented_units
```
Comment (KathleenScriver) [inline test/building_test.rb:98]: Good start.

File: lib/building.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    @units = []
+    @renters = []
+
+  end
+
+  def add_unit(unit)
+    @units << unit
+  end
+
+  def add_renter(renter)
```
Comment (KathleenScriver) [inline lib/building.rb:14]: The Building class didn't ask for a method to add_renters like this. Instead, we already have an array of units, so we can utilize that information to access the renters via the units.

**Comment by abcdefghijohn** [comment]
Thank you for the feedback, I know you wanted us to commit every 15mins or so but I didn't know when and where but now it's clear.

File: lib/building.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    @renters << renter
+  end
+
+  def average_rent
+    monthly_rents.select do |monthly_rent|
+      rent.to_f.sum / units.size
```
Comment (KathleenScriver) [inline lib/building.rb:20]: Talk with this whole method with Graham!

**Pull request closed without merging**